### PR TITLE
[task][buildah] - v0.5: option to provide container registry credentials in a workspace

### PR DESCRIPTION
**Pull request by zroubalik**
Signed-off-by: Zbynek Roubalik <zroubalik@gmail.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/buildah/0.5/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+[Buildah](https://github.com/projectatomic/buildah) build tool. It uses
+Buildah's support for building from
+[`Dockerfile`](https://docs.docker.com/engine/reference/builder/)s, using its
+`buildah bud` command. This command executes the directives in the `Dockerfile`
+to assemble a container image, then pushes that image to a container registry.
+
+## Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/buildah/0.4/buildah.yaml
```
Comment (vinamra28) [inline task/buildah/0.5/README.md:13]:
```suggestion
kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/buildah/0.5/buildah.yaml
```

**Review by vinamra28 — CHANGES_REQUESTED** [review]

File: task/buildah/0.5/README.md (hunk 1 above)
Comment (zroubalik) [reply task/buildah/0.5/README.md]: Good catch, fixed, thanks.

**Comment by vinamra28** [comment]
> Catlin Output
> ```
> FILE: task/buildah/0.5/buildah.yaml
> HINT : Task: tekton.dev/v1beta1 - name: "buildah" is missing a readable display name annotation("tekton.dev/displayName")
> WARN : Step "build" uses image "$(params.BUILDER_IMAGE)" that contains variables; skipping validation
> ```
> 
> Catlin script lint Output
> ```
> ERROR: /usr/bin/shellcheck, [-s sh] failed:
> 
> In buildah-build line 2:
> [[ "$(workspaces.sslcertdir.bound)" == "true" ]] && CERT_DIR_FLAG="--cert-dir $(workspaces.sslcertdir.path)"
> ^-- SC3010: In POSIX sh, [[ ]] is undefined.
> 
> 
> In buildah-build line 3:
> buildah ${CERT_DIR_FLAG} --storage-driver=$(params.STORAGE_DRIVER) bud \
>         ^--------------^ SC2086: Double quote to prevent globbing and word splitting.
>                                           ^----------------------^ SC2046: Quote this to prevent word splitting.
> 
> Did you mean: 
> buildah "${CERT_DIR_FLAG}" --storage-driver=$(params.STORAGE_DRIVER) bud \
> 
> 
> In buildah-build line 4:
>   $(params.BUILD_EXTRA_ARGS) --format=$(params.FORMAT) \
>   ^------------------------^ SC2046: Quote this to prevent word splitting.
>                                       ^--------------^ SC2046: Quote this to prevent word splitting.
> 
> 
> In buildah-build line 5:
>   --tls-verify=$(params.TLSVERIFY) --no-cache \
>                ^-----------------^ SC2046: Quote this to prevent word splitting.
> 
> 
> In buildah-build line 6:
>   -f $(params.DOCKERFILE) -t $(params.IMAGE) $(params.CONTEXT)
>      ^------------------^ SC2046: Quote this to prevent word splitting.
>                              ^-------------^ SC2046: Quote this to prevent word splitting.
>                                              ^---------------^ SC2046: Quote this to prevent word splitting.
> 
> 
> In buildah-build line 7:
> [[ "$(params.SKIP_PUSH)" == "true" ]] && echo "Push skipped" && exit 0
> ^-- SC3010: In POSIX sh, [[ ]] is undefined.
> 
> 
> In buildah-build line 8:
> [[ "$(workspaces.dockerconfig.bound)" == "true" ]] && export DOCKER_CONFIG="$(workspaces.dockerconfig.path)"
> ^-- SC3010: In POSIX sh, [[ ]] is undefined.
>                                                              ^-----------^ SC2155: Declare and assign separately to avoid masking return values.
> 
> 
> In buildah-build line 9:
> buildah ${CERT_DIR_FLAG} --storage-driver=$(params.STORAGE_DRIVER) push \
>         ^--------------^ SC2086: Double quote to prevent globbing and word splitting.
>                                           ^----------------------^ SC2046: Quote this to prevent word splitting.
> 
> Did you mean: 
> buildah "${CERT_DIR_FLAG}" --storage-driver=$(params.STORAGE_DRIVER) push \
> 
> 
> In buildah-build line 10:
>   $(params.PUSH_EXTRA_ARGS) --tls-verify=$(params.TLSVERIFY) \
>   ^-----------------------^ SC2046: Quote this to prevent word splitting.
>                                          ^-----------------^ SC2046: Quote this to prevent word splitting.
> 
> 
> In buildah-build line 11:
>   --digestfile /tmp/image-digest $(params.IMAGE) \
>                                  ^-------------^ SC2046: Quote this to prevent word splitting.
> 
> 
> In buildah-build line 12:
>   docker://$(params.IMAGE)
>            ^-------------^ SC2046: Quote this to prevent word splitting.
> 
> 
> In buildah-build line 13:
> cat /tmp/image-digest | tee $(results.IMAGE_DIGEST.path)
>     ^---------------^ SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.
>                             ^--------------------------^ SC2046: Quote this to prevent word splitting.
> 
> 
> In buildah-build line 14:
> echo "$(params.IMAGE)" | tee $(results.IMAGE_URL.path)
>      ^---------------^ SC2005: Useless echo? Instead of 'echo $(cmd)', just use 'cmd'.
>                              ^-----------------------^ SC2046: Quote this to prevent word splitting.
> 
> For more information:
>   https://www.shellcheck.net/wiki/SC2046 -- Quote this to prevent word splitt...
>   https://www.shellcheck.net/wiki/SC2155 -- Declare and assign separately to ...
>   https://www.shellcheck.net/wiki/SC3010 -- In POSIX sh, [[ ]] is undefined.
> ```

@zroubalik can you please resolve these errors? 😅

**Comment by zroubalik** [comment]
@vinamra28 I am not sure, those errors are completely unrelated to the changes I submitted?

**Comment by vinamra28** [comment]
nevermind @zroubalik, CI is fixed now.
/lgtm
Thanks for your contribution 😄

**Comment by zroubalik** [comment]
So, can we merge this one? :)

**Review by vinamra28 — APPROVED** [review]

**Pull request merged**